### PR TITLE
Integrate new streaming provider Torbox

### PR DESCRIPTION
**Pull request by ragmehos**
### Changes
Integrate new streaming provider Torbox
API docs : https://documenter.getpostman.com/view/29572726/2s9YXo1zX4#intro
Site : https://torbox.app/

### Testing 
* Able to add a magnet link
* After download is finished, able to generate a URL and stream it. 

### Note
* This change deviates a bit from other implementations, after adding the magnet link, it doesn't wait to check for the state, it just raises a "Torrent did not reach downloaded status" - that is cause the API is a bit rudimentary, adding the magnet link does not return any status or id, it just says success, so let the user retry. I can however change this to get the torrent and check, but thought this is ok for now. 
* Referral link also not added, there is an explicit POST request made for referral (to the link "https://db.torbox.app/rest/v1/referrals") and hence, not adding a referral link for sign up.
* There is no API for figuring out if a torrent is cached, so it's similar to pikpak where we can only check with the list of torrents on Torbox.

### CODE REVIEW

File: resources/js/config_script.js [hunk 1]
```diff
@@ -8,7 +8,8 @@ const providerSignupLinks = {
     offcloud: 'https://offcloud.com/?=9932cd9f',
     realdebrid: 'http://real-debrid.com/?id=9490816',
     debridlink: 'https://debrid-link.com/id/kHgZs',
-    alldebrid: 'https://alldebrid.com/?uid=3ndha&lang=en'
+    alldebrid: 'https://alldebrid.com/?uid=3ndha&lang=en',
+    torbox: 'https://torbox.app/'
```
Comment (mhdzumair) [inline resources/js/config_script.js:12]: You're welcome to add your own referral link.

File: streaming_providers/torbox/utils.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    # Check if the torrent already exists
+    torrent_info = torbox_client.get_available_torrent(info_hash)
+    if torrent_info:
+        if torrent_info["download_finished"] is True and torrent_info["download_present"] is True:
+            file_id = select_file_id_from_torrent(torrent_info, filename)
+            response = torbox_client.create_download_link(torrent_info.get("id"), file_id)
+            return response["data"]
+    else:
+        # If torrent doesn't exist, add it
+        _ = torbox_client.add_magent_link(magnet_link)
```
Comment (mhdzumair) [inline streaming_providers/torbox/utils.py:28]:
```suggestion
        torbox_client.add_magent_link(magnet_link)
```

**Review by mhdzumair — CHANGES_REQUESTED** [review]
Overall code is looks good to me. Raising torrent_not_downloaded error also fine. Stremio query two times when user click. So it should be fine and user can retry few times.

**Comment by ragmehos** [comment]
> Overall code is looks good to me. Raising torrent_not_downloaded error also fine. Stremio query two times when user click. So it should be fine and user can retry few times.

I have also requested the dev to provide an API to check if a torrent is cached or not, I can make the changes once that API is available and make this code path a bit better and faster. 

It's also good to know that stremio queries it two times, I was confused as to why I was always getting multiple queries to the same endpoints :D

File: resources/js/config_script.js (hunk 1 above)
Comment (ragmehos) [reply resources/js/config_script.js:12]: Sure, will do it once I figure out a way to maybe get the referral into the URL itself :) Thanks.

**Review by mhdzumair — APPROVED** [review]

**Pull request merged**